### PR TITLE
Allow docs/Makefile to be portable

### PR DESCRIPTION
**Pull request by daniellimws**
Currently docs/Makefile is configured to download only the Linux version of Conda. This commit allows it to detect the operating system and download the corresponding Conda installer.

Additionally, `make` uses the sh shell which does not support the `source` command. This causes the commands that use `source` to fail on some systems that do not have the sh shell linked to something else (say dash).

I only added the MacOSX installer because I don't have a Windows machine to test on. Similarly, I've added detection for Cygwin and MINGW environments but have yet to test them as well, although I think they should work as expected.

Signed-off-by: Daniel Lim Wee Soong <weesoong.lim@gmail.com>

### CODE REVIEW

File: docs/Makefile [hunk 1]
```diff
@@ -5,11 +5,26 @@ MAKEDIR := $(dir $(lastword $(MAKEFILE_LIST)))
...
 # You can set these variables from the command line.
 SPHINXOPTS      =
-SPHINXBUILD     = [ -e env/bin/activate ] && source env/bin/activate; sphinx-build
-SPHINXAUTOBUILD = [ -e env/bin/activate ] && source env/bin/activate; sphinx-autobuild
+SPHINXBUILD     = [ -e env/bin/activate ] && . env/bin/activate; sphinx-build
+SPHINXAUTOBUILD = [ -e env/bin/activate ] && . env/bin/activate; sphinx-autobuild
 SPHINXPROJ      = SymbiFlowV2X
 SOURCEDIR       = .
 BUILDDIR        = _build
+OSFLAG 				:=
```
Comment (mithro) [inline docs/Makefile:13]: Indenting on the `:=` is a bit weird.

File: docs/Makefile [hunk 2]
```diff
@@ -21,18 +36,18 @@ livehtml:
...
 env:
 	rm -rf env
-	make env/Miniconda3-latest-Linux-x86_64.sh
-	./env/Miniconda3-latest-Linux-x86_64.sh -p $(PWD)/env -b -f
-	source env/bin/activate; conda config --system --add envs_dirs $(PWD)/env/envs
-	source env/bin/activate; conda config --system --add pkgs_dirs $(PWD)/env/pkgs
-	source env/bin/activate; conda env update --name base --file ./environment.yml
+	make env/Miniconda3
+	./env/Miniconda3-latest-$(OSFLAG)-x86_64.sh -p $(PWD)/env -b -f
+	. env/bin/activate; conda config --system --add envs_dirs $(PWD)/env/envs
```
Comment (mithro) [inline docs/Makefile:48]: Why the change from `source` to `.`?

File: docs/Makefile (hunk 1 above)
Comment (daniellimws) [reply docs/Makefile:13]: You're right, will change this.

File: docs/Makefile (hunk 2 above)
Comment (daniellimws) [reply docs/Makefile:48]: Using `source` might cause this to fail on some systems (it failed on my Ubuntu 18.04) because `make` runs on the `sh` shell, that does not support `source`. It would only work for those that have `sh` being actually the `dash` shell installed in the system (this is what I found online).

File: docs/Makefile [hunk 3]
```diff
@@ -10,7 +10,7 @@ SPHINXAUTOBUILD = [ -e env/bin/activate ] && . env/bin/activate; sphinx-autobuil
 SPHINXPROJ      = SymbiFlowV2X
 SOURCEDIR       = .
 BUILDDIR        = _build
-OSFLAG 				:=
+OSFLAG          =
```
Comment (mithro) [inline docs/Makefile:13]: `SHELL = /bin/bash` maybe?

File: docs/Makefile [hunk 4]
```diff
@@ -21,15 +37,15 @@ livehtml:
 .PHONY: help livehtml Makefile
 
 
-env/Miniconda3-latest-Linux-x86_64.sh:
+env/Miniconda3:
```
Comment (mithro) [inline docs/Makefile:40]:
```suggestion
env/Miniconda3-latest-$(OSFLAG)-x86_64.sh:
```

File: docs/Makefile [hunk 5]
```diff
@@ -21,15 +37,15 @@ livehtml:
...
-	wget https://repo.anaconda.com/miniconda/Miniconda3-latest-Linux-x86_64.sh -O env/Miniconda3-latest-Linux-x86_64.sh
-	chmod a+x env/Miniconda3-latest-Linux-x86_64.sh
+	wget https://repo.anaconda.com/miniconda/Miniconda3-latest-$(OSFLAG)-x86_64.sh -O env/Miniconda3-latest-$(OSFLAG)-x86_64.sh
+	chmod a+x env/Miniconda3-latest-$(OSFLAG)-x86_64.sh
 
 env:
 	rm -rf env
-	make env/Miniconda3-latest-Linux-x86_64.sh
-	./env/Miniconda3-latest-Linux-x86_64.sh -p $(PWD)/env -b -f
+	make env/Miniconda3
```
Comment (mithro) [inline docs/Makefile:47]:
```suggestion
	make env/Miniconda3-latest-$(OSFLAG)-x86_64.sh
```

File: docs/Makefile (hunk 4 above)
Comment (mithro) [reply docs/Makefile:40]: Otherwise this becomes a virtual rule that will *always* run.

**Comment by daniellimws** [comment]
I forgot to mention there's another issue with running `make env` on MacOS. I get the following error:

```
...
Preparing transaction: done
Executing transaction: done
installation finished.
source env/bin/activate; conda config --system --add envs_dirs /Users/daniellimws/python-symbiflow-v2x/docs/env/envs
source env/bin/activate; conda config --system --add pkgs_dirs /Users/daniellimws/python-symbiflow-v2x/docs/env/pkgs
source env/bin/activate; conda env update --name base --file ./environment.yml
Collecting package metadata (repodata.json): done
Solving environment: failed

ResolvePackageNotFound:
  - netlistsvg
  - yosys

make: *** [env] Error 1
```

This happens because the SymbiFlow channel on Conda does not have MacOS builds for the yosys and netlistsvg packages. My workaround is to temporarily delete these 2 dependencies from environment.yml and manually install them to my system.

I suppose it is not worth the hassle to add MacOS builds to the conda channel so is it fine to leave it like this for now?

**Comment by mithro** [comment]
FYI - @PiotrZierhoffer is currently working on adding yosys and netlistsvg packages to conda for Mac and Windows.

**Comment by daniellimws** [comment]
@mithro Ok to merge this?
